code/modules/urist/modules/GOAI/integrations/base_commander/mob_commander/combat_commander/orders.dm

```diff
@@ -8,8 +8,8 @@
 	if(!(M?.brain))
 		return
 
-	var/datum/memory/created_mem = M.brain.SetMemory(MEM_WAYPOINT_IDENTITY, Trg, PLUS_INF)
-	var/atom/waypoint = created_mem?.val
+	var/datum/memory/created_mem = M.brain.SetMemory(MEM_WAYPOINT_IDENTITY, weakref(Trg), PLUS_INF)
+	var/atom/waypoint = resolve_weakref(created_mem?.val)
```

Comment:
Could just use Trg for this, I think?

Reply:
I did think so, but I wasn't sure if you'd deliberately done this for some sort of test or to validate behaviour. I'll switch it over